transport/http/outbound.go

```diff
@@ -386,16 +390,24 @@ func (o *Outbound) withCoreHeaders(req *http.Request, treq *transport.Request, t
 		req.Header.Set(EncodingHeader, encoding)
 	}
 
+	req.Header.Set(AcceptsBothResponseErrorHeader, AcceptTrue)
+
 	return req
 }
 
-func getYARPCErrorFromResponse(response *http.Response) error {
-	contents, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return yarpcerrors.Newf(yarpcerrors.CodeInternal, err.Error())
-	}
-	if err := response.Body.Close(); err != nil {
-		return yarpcerrors.Newf(yarpcerrors.CodeInternal, err.Error())
+func getYARPCErrorFromResponse(response *http.Response, bothResponseError bool) error {
+	var contents string
+	if bothResponseError {
+		contents = response.Header.Get(ErrorMessageHeader)
+	} else {
+		contentsBytes, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return yarpcerrors.Newf(yarpcerrors.CodeInternal, err.Error())
+		}
+		contents = string(contentsBytes)
+		if err := response.Body.Close(); err != nil {
+			return yarpcerrors.Newf(yarpcerrors.CodeInternal, err.Error())
+		}
 	}
 	// use the status code if we can't get a code from the headers
 	code := statusCodeToBestCode(response.StatusCode)
```

Comment:
can we extract this into a function?  "getErrorStatusCode"?

Reply:
`statusCodeToBestCode` is a function